Intersect.Client.Core/Interface/Game/TargetContextMenu.cs

```diff
@@ -116,6 +132,21 @@ public void ToggleHidden(object? target)
         }
     }
 
+
+    private void TryShowTargetButton(bool shouldShow)
+    {
+        _targetNameMenuItem.SetText(shouldShow ? _entity.Name : string.Empty);
+
+        if (shouldShow && !Children.Contains(_targetNameMenuItem))
+        {
+            Children.Insert(0, _targetNameMenuItem);
```

Comment:
do the following pseudocode instead to account for all of the cases exhaustively

```
if (shouldShow) {
   indexOf
   if (indexOf > 0) remove it // just remove, it's inserted in the correct location on the next line
   if (indexOf != 0) insert at 0 // (this takes into account both cases where it's not present at all, and is present but in the wrong place
}
```